### PR TITLE
ci: Add initial scale test workflow for bootstrapping

### LINKED ISSUE #27749: Bootstrap GHA for scale-testing
As part of https://github.com/cilium/cilium/issues/27196 we are going to implement GithubAction for scale-testing Cilium.

This issue is for the first step of implementation:
- Creating a new GithubAction
- Spinning small (2-3 nodes) cluster using kOps
- Tearing down cluster.

First, we want to ensure that our pipeline does not leak any resources and cleans up everything, before processing with larger clusters and actual scale-testing.

### PR DESCRIPTION
**Pull request by learnitall**
Please ensure your pull request adheres to the following guidelines:

- [X] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [X] All code is covered by unit and/or runtime tests where feasible.
- [X] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [X] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [X] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [X] Provide a title or release-note blurb suitable for the release notes.
- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [X] Thanks for contributing!

This commit acts as an initial effort to add automated scale testing to Cilium, as described in https://github.com/cilium/design-cfps/pull/11. The workflow added as part of this commit is meant to lay down a foundation that can be built upon, with the intent that it will be iteratively developed on to carry out the CFP. The workflow currently performs the following:

* Create a two-node GCE cluster using kops
* Run a bare-bones ClusterLoader2 configuration
* Cleanup the cluster

The following next steps are a WIP:

* Expand on the ClusterLoader2 configuration
* Increase the number of nodes in the cluster
* Export results from ClusterLoader2
* Allow for manual runs using WorkflowDispatch

Note: ClusterLoader2 is currently pulled from a fork containing changes that are intended to be merged upstream in the coming week.

Fixes: https://github.com/cilium/cilium/issues/27749

```release-note
Add initial, in-progress workflow for automated scale testing
```

### CODE REVIEW

File: .github/workflows/scale-test-gce.yaml [hunk 1]
```diff
@@ -0,0 +1,230 @@
+name: Scale Test GCE
+
+on:
+  schedule:
+    - cron: '39 0 * * 1,4'
+  push:
```
Comment (brlbil) [inline .github/workflows/scale-test-gce.yaml]: I think this trigger needs to be removed before merging.

**Review by brlbil — CHANGES_REQUESTED** [review]

File: .github/workflows/scale-test-gce.yaml (hunk 1 above)
Comment (learnitall) [reply .github/workflows/scale-test-gce.yaml]: For sure, I left that in for testing. I just removed it.

**Review by brlbil — APPROVED** [review]
LGTM

**Comment by learnitall** [comment]
Before doing a force push to remove items from the workflow that were needed for testing (a push trigger), this was the last successful run: https://github.com/cilium/cilium/actions/runs/6424076662/job/17443936075. Since this PR just adds a new workflow file and doesn't make any other changes, I don't believe it needs the full test suite. Therefore, I'm going to mark this as ready to merge.

**Pull request merged**